src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -680,9 +673,63 @@ namespace shogun
 		SGMatrix<T>
 		cholesky_factor(const SGMatrix<T>& A, const bool lower = true)
 		{
+			REQUIRE(
+			    A.num_rows == A.num_cols,
+			    "Matrix dimensions (%dx%d) are not square\n", A.num_rows,
+			    A.num_cols);
 			return infer_backend(A)->cholesky_factor(A, lower);
 		}
 
+		/**
+		 * Updates the Cholesky factorization \f$A = L L^{*}\f$ with a rank one
+		 * term in-place.
+		 * If A = LL^T before the update, then after it
+		 * LL^{*} = A + alpha * b b^{*}
```

Comment:
Quite nitpicking: perhaps a different name for the L as they are not the same as the ones above. In wikipedia they use tilde analogously for A. It is completely subjective I think, whether we want the doc to be more math correct, or just reflect what happens in the code. I am happy either way :-)